File: src/SlimShader/Chunks/Shex/SyncFlags.cs
using System;

namespace SlimShader.Chunks.Shex
{
	[Flags]
	public enum SyncFlags
	{
		None,

		ThreadsInGroup = 1,
		SharedMemory = 2,
		UnorderedAccessViewGroup = 4,
		UnorderedAccessViewGlobal = 8
	}
}

File: src/cpp/Source/Chunks/Shex/SyncFlags.cpp
#include "PCH.h"
#include "SyncFlags.h"

#include "Decoder.h"

using namespace std;
using namespace SlimShader;

string SlimShader::ToString(SyncFlags value)
{
	string result;

	if (HasFlag(value, SyncFlags::UnorderedAccessViewGlobal))
		result += "_uglobal";
	if (HasFlag(value, SyncFlags::UnorderedAccessViewGroup))
		result += "_ugroup";
	if (HasFlag(value, SyncFlags::SharedMemory))
		result += "_g";
	if (HasFlag(value, SyncFlags::ThreadsInGroup))
		result += "_t";

	return result;
}